src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.ts

```diff
@@ -0,0 +1,113 @@
+import { ReactWrapper, ShallowWrapper, mount, shallow } from "enzyme";
+import { DOM, createElement } from "react";
+
+import { Carousel, CarouselProps, Image } from "../Carousel";
+import { CarouselItem } from "../CarouselItem";
+
+describe("Carousel", () => {
+    let images: Image[];
+    describe("shallow component", () => {
+        let output: ShallowWrapper<CarouselProps, any>;
+        let carouselWrapper: ShallowWrapper<CarouselProps, any>;
+        describe("with multiple images", () => {
+            beforeEach(() => {
+                images = [
+                    { url: "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" },
+                    { url: "https://www.google.com/images/nav_logo242.png" }
+                ];
+                output = shallow(createElement(Carousel, { images }));
+                carouselWrapper = output.find("div.carousel-inner");
+            });
+
+            it("renders a div of class carousel with a div child of class carousel-inner", () => {
+
+                expect(output).toMatchStructure(DOM.div({ className: "carousel" },
+                    DOM.div({ className: "carousel-inner" })));
+            });
+
+            it("renders a wrapper with the same number of children as the images", () => {
+
+                expect(carouselWrapper.children().length).toBe(images.length);
+            });
+
+            it("renders a wrapper with the first child active", () => {
```

Comment:
Since the main element is going to have more children in the future besides the 'wrapper', perhaps I should stop calling them children in the description. Images perhaps.